### PR TITLE
add deprecation warning for share module

### PR DESCRIPTION
**Pull request by Ansalibrahim**
Please Check :-
- https://github.com/facebook/react-native/issues/23313
- https://github.com/react-native-community/react-native-simple-share

## Summary
As a part of [lean core](https://github.com/facebook/react-native/issues/23313) , I have stripped out the share module in react-native to [react-native-simple-share](https://github.com/react-native-community/react-native-simple-share). So I have added a deprecation warning in Share module in this repo.

## Changelog
- Deprecating share  module
   - added a console.warn statement indicating that the module is deprecated

## Test Plan
Only code change is `console.warn(DEPRECATION_WARNING);`, it will show a warning : 

> Warning: Share package is deprecated in favour of the lean core (https://github.com/facebook/react-native/issues/23313). Use https://github.com/react-native-community/react-native-simple-share for the exact same feature set or use https://github.com/react-native-community/react-native-share which has a rich feature set

### CODE REVIEW

File: Libraries/Share/Share.js [hunk 1]
```diff
@@ -27,8 +27,14 @@ type Options = {
   subject?: string,
 };
 
+const DEPRECATION_WARNING = 'Warning: Share package is deprecated in favour of the lean core (https://github.com/facebook/react-native/issues/23313). Use https://github.com/react-native-community/react-native-simple-share for the exact same feature set or use https://github.com/react-native-community/react-native-share which has a rich feature set';
+
 class Share {
   /**
+   * Share is deprecated in favour of lean core 
+   * https://github.com/facebook/react-native/issues/23313 and
```
[the remark this answers is not in the document]
Comment (Ansalibrahim) [reply Libraries/Share/Share.js]: removed the line

[the remark this answers is not in the document]
Comment (Ansalibrahim) [reply Libraries/Share/Share.js]: resolved

File: Libraries/Share/Share.js [hunk 2]
```diff
@@ -27,8 +27,14 @@ type Options = {
...
 };
 
+const DEPRECATION_WARNING = 'Warning: Share package is deprecated in favour of the lean core (https://github.com/facebook/react-native/issues/23313). Use https://github.com/react-native-community/react-native-simple-share for the exact same feature set or use https://github.com/react-native-community/react-native-share which has a rich feature set';
+
 class Share {
   /**
+   * Share is deprecated in favour of lean core 
+   * https://github.com/facebook/react-native/issues/23313 and 
+   * https://github.com/react-native-community/react-native-simple-share
+   *
```
[the remark this answers is not in the document]
Comment (Ansalibrahim) [reply Libraries/Share/Share.js]: resolved

[the remark this answers is not in the document]
Comment (Ansalibrahim) [reply Libraries/Share/Share.js]: resolved

File: Libraries/Share/Share.js [hunk 3]
```diff
@@ -27,8 +27,14 @@ type Options = {
   subject?: string,
 };
 
+const DEPRECATION_WARNING = 'Warning: Share package is deprecated in favour of the lean core (https://github.com/facebook/react-native/issues/23313). Use https://github.com/react-native-community/react-native-simple-share for the exact same feature set or use https://github.com/react-native-community/react-native-share which has a rich feature set';
```
[the remark this answers is not in the document]
Comment (Ansalibrahim) [reply Libraries/Share/Share.js]: resolved

File: Libraries/Share/Share.js [hunk 4]
```diff
@@ -27,8 +27,14 @@ type Options = {
...
+
 class Share {
   /**
+   * Share is deprecated in favour of lean core
```
[the remark this answers is not in the document]
Comment (Ansalibrahim) [reply Libraries/Share/Share.js]: resolved

[the remark this answers is not in the document]
Comment (Ansalibrahim) [reply Libraries/Share/Share.js]: resolved

**Comment by cpojer** [comment]
Thanks for your help. I think tis is not quite ready yet and we probably won't remove the share module for quite a while so I'm going to close this PR for now.

**Pull request closed without merging**